### PR TITLE
docs(identity): add keycloak 23 and remove keycloak 21 from identity …

### PR DESCRIPTION
**Pull request by maryarm**
…supported environment

## Description

Relevant issue: [2407](https://github.com/camunda-cloud/identity/issues/2407)

## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [x] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [ ] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [x] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Review by christinaausley — APPROVED** [review]

**Comment by maryarm** [comment]
> > 👋 🤖 🤔 Hello! Did you make your changes in all the right places?
> > These files were changed only in _docs/_. You might want to duplicate these changes in _versioned_docs/version-8.4/_.
> > 
> > * docs/reference/supported-environments.md
> > * docs/self-managed/identity/getting-started/install-identity.md
> > * docs/self-managed/identity/troubleshooting/troubleshoot-identity.md
> > * docs/self-managed/identity/user-guide/configuration/configure-external-identity-provider.md
> > * docs/self-managed/identity/user-guide/configuration/connect-to-an-existing-keycloak.md
> > 
> > You may have done this intentionally, but we wanted to point it out in case you didn't. You can read more about the versioning within our docs in our [documentation guidelines](https://github.com/camunda/camunda-docs/blob/c666b7e5e35bfe31468ce4ac679ea7b62ab86d73/howtos/documentation-guidelines.md#versions).
> 
> @maryarm, your PR is currently making changes for 8.5. Is this your intention? Or would you like to update the 8.4 docs as well? Based on the linked issue, I think you want to add it to 8.4

Hi @christinaausley, the changes are for 8.5. The intension of this PR is to have the documentation ready sooner, it let the customers to be aware of the changes of supported environment of identity sooner (keycloak 21 is not supported any more, and we have added the support for keycloak 23).

File: docs/reference/supported-environments.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ Requirements for the components can be seen below:
...
 | Operate     | OpenJDK 17+  | Elasticsearch 8.9+<br/>Amazon OpenSearch 2.5.x                                                                                                                       |
 | Tasklist    | OpenJDK 17+  | Elasticsearch 8.9+<br/>Amazon OpenSearch 2.5.x                                                                                                                       |
-| Identity    | OpenJDK 17+  | Keycloak 21.x, 22.x<br/>PostgreSQL 14.x, 15.x (required for [certain features](/self-managed/identity/deployment/configuration-variables.md#database-configuration)) |
+| Identity    | OpenJDK 17+  | Keycloak 22.x, 23.x<br/>PostgreSQL 14.x, 15.x (required for [certain features](/self-managed/identity/deployment/configuration-variables.md#database-configuration)) |
 | Optimize    | OpenJDK 17+  | Elasticsearch 8.9+                                                                                                                                                   |
 | Web Modeler | -            | Keycloak 21.x, 22.x<br/>PostgreSQL 13.x, 14.x, 15.x, Amazon Aurora PostgreSQL 13.x, 14.x, 15.x (other database systems are currently not supported)                  |
```
Comment (Ben-Sheppard) [inline docs/reference/supported-environments.md:47]: 🔧 this line for Web Modeler can also be changed I believe (@wollefitz please correct me if I'm wrong)

**Review by Ben-Sheppard — COMMENTED** [review]
Just one comment on keeping the web modeler versions in sync too :)

**Comment by maryarm** [comment]
> Are we still planning to release this tomorrow? I saw the issue has an `alpha-1` label.

That would be nice, should I merge now?

File: docs/reference/supported-environments.md (hunk 1 above)
Comment (maryarm) [reply docs/reference/supported-environments.md:47]: As I informed web modeler before, they are going to take care of their changes in a separate PR.

**Pull request merged**